pkg/plugins/golang/v3/api.go

```diff
@@ -142,6 +143,11 @@ func (p *createAPISubcommand) InjectResource(res *resource.Resource) error {
 
 	// In case we want to scaffold a resource API we need to do some checks
 	if p.options.DoAPI {
+		// Generating the API for a core resource is forbidden.
+		if strings.HasPrefix(p.resource.Path, "k8s.io") {
+			return errors.New("scaffolding API for core resources is not supported")
+		}
```

Comment:
WDYT about if is core-type then we allow to do the scaffold and just update the PROJECT file?

Reply:
Do I understand correctly you'd like to ignore the `--resource` flag completely for the core types and silently set it to false?

Reply:
@camilamacedo86 I believe as a first iteration `create api` may fail when for a core resource. If the handling of the `--resource` flag changes in the future it will be a non-breaking change anyway if it fails now.

Reply:
We can create api for a core type however that will be added to the project domain and is required to change manually. So, the correct approach IMO is:

- create API: check if is a core-type and do not scaffold the resource and populate accordingly the PROJECT file
- webhook API: check if we are trying to create a webhook for a core type, if yes then call create API for that to do the same above and allow the default scaffold occurs.

Reply:
I am not sure. You added `return errors.New("scaffolding API for core resources is not supported")`